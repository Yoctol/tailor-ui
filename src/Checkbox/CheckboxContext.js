import { createContext } from 'react';

const { Provider, Consumer } = createContext({
  _onChange: null,
  _isChecked: null,
});

export { Provider, Consumer };
