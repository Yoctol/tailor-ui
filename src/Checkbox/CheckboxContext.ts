import { ChangeEvent, createContext } from 'react';

export type Direction = 'horizontal' | 'vertical';

const CheckboxContext = createContext<{
  _onChange?: (event: ChangeEvent<HTMLInputElement>, value: string) => void;
  _isChecked?: (value: string) => boolean;
  direction: Direction;
}>({
  direction: 'horizontal',
});

export default CheckboxContext;
