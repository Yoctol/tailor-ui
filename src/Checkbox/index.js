import React from 'react';
import PropTypes from 'prop-types';
import RcCheckbox from 'rc-checkbox';
import styled from 'styled-components';
import { themeGet, width, height, color, borderColor } from 'styled-system';
import { ifProp } from 'styled-tools';

import theme from '../theme';

const CheckboxWrapper = styled.span`
  .rc-checkbox {
    display: inline-block;
    position: relative;
    outline: none;
    line-height: 1;
  }

  .rc-checkbox-inner {
    display: inline-block;
    position: relative;
    top: 0;
    left: 0;
    border: 1px solid
      ${ifProp(
        'disabled',
        themeGet('colors.gray.8'),
        props => themeGet(`colors.${props.borderColor}`) || props.borderColor
      )};
    border-radius: ${ifProp('circle', props => +props.height / 2, '3')}px;
    background-color: #fff;

    ${width};
    ${height};
  }

  .rc-checkbox-inner::after {
    content: ' ';
    display: table;
    position: absolute;
    top: ${props => (+props.height - 11) / 2}px;
    left: ${props => (+props.height - 7) / 2}px;
    width: 5px;
    height: 8px;
    border-top: 0;
    border-left: 0;
    border-radius: ${themeGet('radii.1')};
    transform: rotate(45deg);
  }

  .rc-checkbox-input {
    position: absolute;
    z-index: 9999;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    opacity: 0;
    cursor: ${ifProp('disabled', 'default', 'pointer')};

    ${width};
    ${height};
  }

  .rc-checkbox-checked .rc-checkbox-inner {
    ${borderColor};
    ${color};
  }

  .rc-checkbox-checked .rc-checkbox-inner::after {
    content: ' ';
    display: table;
    position: absolute;
    top: ${props => (+props.height - 11) / 2}px;
    left: ${props => (+props.height - 7) / 2}px;
    border: 2px solid #fff;
    border-top: 0;
    border-left: 0;
    transform: rotate(45deg);
  }

  .rc-checkbox-disabled .rc-checkbox-inner::after {
    border-color: #f3f3f3;
  }

  .rc-checkbox-disabled.rc-checkbox-checked .rc-checkbox-inner::after {
    border-color: #ccc;
  }

  .rc-checkbox-disabled .rc-checkbox-inner {
    border-color: #d9d9d9;
    background-color: #f3f3f3;
  }
`;

CheckboxWrapper.propTypes = {
  circle: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  ...borderColor.propTypes,
  ...color.propTypes,
  ...width.propTypes,
  ...height.propTypes,
};

CheckboxWrapper.defaultProps = {
  theme,
  borderColor: 'primary',
  bg: 'primary',
  height: 19,
  width: 19,
};

const Checkbox = ({
  circle,
  disabled,
  borderColor: _borderColor,
  bg: _bg,
  width: _width,
  height: _height,
  ...props
}) => (
  <CheckboxWrapper
    circle={circle}
    disabled={disabled}
    borderColor={_borderColor}
    bg={_bg}
    width={_width}
    height={_height}
  >
    <RcCheckbox disabled={disabled} {...props} />
  </CheckboxWrapper>
);

Checkbox.propTypes = {
  circle: PropTypes.bool,
  disabled: PropTypes.bool,
  ...borderColor.propTypes,
  ...color.propTypes,
  ...width.propTypes,
  ...height.propTypes,
};

Checkbox.defaultProps = {
  circle: false,
  disabled: false,
};

export default Checkbox;
