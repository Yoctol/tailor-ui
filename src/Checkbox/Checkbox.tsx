import React, { ChangeEvent, FunctionComponent, useContext } from 'react';
import styled, { css } from 'styled-components';

import tag from 'utils/CleanTag';

import Space from '../Grid/Space';

import CheckboxContext, { Direction } from './CheckboxContext';
import CheckboxGroup from './CheckboxGroup';

const CheckboxWrapper = styled.span`
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
`;

const CheckboxInner = styled.span`
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: ${p => p.theme.borders.base};
  border-radius: 2px;
  border-color: ${p => p.theme.colors.gray400};
  background-color: ${p => p.theme.colors.light};
  color: ${p => p.theme.colors.gray700};

  ${p => p.theme.transition};

  &::after {
    content: '';
    position: absolute;
    top: 1px;
    left: 5px;
    width: 5px;
    height: 10px;
    border: solid white;
    border-width: 0 2px 2px 0;
    opacity: 0;
    transform: rotate(45deg) scale(0);
    ${p => p.theme.transition};
  }
`;

const StyledCheckbox = styled.input.attrs({
  type: 'checkbox',
})`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;

  &:checked + ${CheckboxInner /* sc-selector */} {
    border-color: ${p => p.theme.colors.primary};
    background-color: ${p => p.theme.colors.primary};
  }

  &:checked + ${CheckboxInner /* sc-selector */}::after {
    opacity: 1;
    transform: rotate(45deg) scale(1);
  }
`;

const getMarginPosition = ({ direction }: { direction: Direction }) =>
  direction === 'horizontal' ? 'margin-left' : 'margin-top';

interface IRadioLabelBaseProps {
  disabled: boolean;
}

const CheckboxLabelBase = styled(tag.label)<IRadioLabelBaseProps>`
  display: inline-flex;
  align-items: center;
  font-size: ${p => p.theme.fontSizes.base};
  cursor: ${p => (p.disabled ? 'not-allowed' : 'pointer')};

  & + & {
    ${getMarginPosition /* sc-prop */}: ${p => p.theme.space[2]};
  }

  ${p =>
    p.disabled
      ? css`
          opacity: 0.5;
        `
      : css`
          &:hover ${CheckboxInner /* sc-selector */} {
            border-color: ${p.theme.colors.primary};
          }
        `};
`;

const CheckboxLabel = styled(CheckboxLabelBase)`
  /* stylelint-disable-next-line no-descending-specificity */
  & + ${CheckboxLabelBase /* sc-selector */} {
    ${getMarginPosition /* sc-prop */}: ${p => p.theme.space[2]};
  }
`;

export interface ICheckboxProps {
  /**
   * Specifies whether the checkbox is selected
   */
  checked?: boolean;
  /**
   * Specifies the initial state: whether or not the checkbox is selected.
   */
  defaultChecked?: boolean;
  /**
   * Disable Checkbox
   */
  disabled?: boolean;
  /**
   * The callback function that is triggered when the state changes
   */
  onChange?: (event: ChangeEvent) => void;
  /**
   * The value of checkbox
   */
  value?: string;
}

const Checkbox: FunctionComponent<ICheckboxProps> & {
  Group: typeof CheckboxGroup;
} = ({
  children,
  disabled = false,
  checked,
  defaultChecked,
  onChange,
  value,
  ...props
}) => {
  const { _onChange, _isChecked, direction } = useContext(CheckboxContext);

  return (
    <CheckboxLabel disabled={disabled} direction={direction} {...props}>
      <CheckboxWrapper>
        <StyledCheckbox
          disabled={disabled}
          checked={_isChecked && value ? _isChecked(value) : checked}
          defaultChecked={defaultChecked}
          onChange={event => {
            if (onChange) {
              onChange(event);
            }
            if (_onChange && value) {
              _onChange(event, value);
            }
          }}
        />
        <CheckboxInner />
      </CheckboxWrapper>
      <Space px="2">{children}</Space>
    </CheckboxLabel>
  );
};

Checkbox.Group = CheckboxGroup;

export default Checkbox;
