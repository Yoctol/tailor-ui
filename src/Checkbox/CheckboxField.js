import PropTypes from 'prop-types';
import React from 'react';
import { Set } from 'react-powerplug';

import FormField from '../Form/FormField';
import Hint from '../Form/Hint';
import Label from '../Form/Label';
import Space from '../Grid/Space';

import Checkbox from '.';

let seed = 0;

const getUuid = () => {
  seed += 1;
  return `yoctol_checkbox_${Date.now()}_${seed}`;
};

const CheckboxField = ({
  children,
  success,
  warning,
  error,
  message,
  ...otherProps
}) => {
  const id = getUuid();
  return (
    <FormField success={success} warning={warning} error={error}>
      <Checkbox id={id} {...otherProps} />
      <Label htmlFor={id}>{children}</Label>
      {(success || warning || error) && <Hint>{message}</Hint>}
    </FormField>
  );
};

CheckboxField.propTypes = {
  /**
   * Specifies whether the checkbox is selected
   */
  checked: PropTypes.bool,
  /**
   * The label of checkbox field
   */
  children: PropTypes.string.isRequired,
  /**
   * Disable checkbox
   */
  disabled: PropTypes.bool,
  /**
   * Set the checkbox group status to error
   */
  error: PropTypes.bool,
  /**
   * The message will show when success or warning or error is true
   */
  message: PropTypes.string,
  /**
   * Set the checkbox group status to success
   */
  success: PropTypes.bool,
  /**
   * Set the checkbox group status to warning
   */
  warning: PropTypes.bool,
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
};

CheckboxField.defaultProps = {
  disabled: false,
  checked: false,
  error: false,
  message: '',
  success: false,
  warning: false,
  onChange: () => {},
};

export const CheckboxFieldGroup = ({
  label: groupLabel,
  values: groupValues,
  initialValues,
  options,
  onChange,
  success,
  warning,
  error,
  message,
  ...otherProps
}) => (
  <Set initial={initialValues || []} onChange={onChange}>
    {({ add, remove, has }) => (
      <FormField success={success} warning={warning} error={error}>
        {groupLabel && <Label>{groupLabel}</Label>}
        {options.map(({ label, value, disabled = false }) => {
          const id = getUuid();
          return (
            <Space mt={2} mb={1} key={label}>
              <Checkbox
                id={id}
                checked={groupValues ? groupValues.includes(value) : has(value)}
                disabled={disabled}
                onChange={event => {
                  const { checked } = event.target;
                  if (groupValues) {
                    if (groupValues.includes(value) && !checked) {
                      onChange(groupValues.filter(val => val !== value));
                    }
                    if (!groupValues.includes(value) && checked) {
                      onChange(groupValues.push(value));
                    }
                  } else {
                    if (has(value) && !checked) {
                      remove(value);
                    }
                    if (!has(value) && checked) {
                      add(value);
                    }
                  }
                }}
                {...otherProps}
              />
              <Label htmlFor={id}>{label}</Label>
            </Space>
          );
        })}
        {(success || warning || error) && <Hint>{message}</Hint>}
      </FormField>
    )}
  </Set>
);

CheckboxFieldGroup.displayName = 'CheckboxField.Group';

CheckboxFieldGroup.propTypes = {
  /**
   * Set the checkbox field group status to error
   */
  error: PropTypes.bool,
  /**
   * To set the initial values
   */
  initialValues: PropTypes.arrayOf(PropTypes.string),
  /**
   * The label of checkbox field group
   */
  label: PropTypes.string,
  /**
   * The message will show when success or warning or error is true
   */
  message: PropTypes.string,
  /**
   * Specifies options
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
      disabled: PropTypes.bool,
    })
  ).isRequired,
  /**
   * Set the checkbox field group status to success
   */
  success: PropTypes.bool,
  /**
   * Used for setting the currently selected value
   */
  values: PropTypes.arrayOf(PropTypes.string),
  /**
   * Set the checkbox field group status to warning
   */
  warning: PropTypes.bool,
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
};

CheckboxFieldGroup.defaultProps = {
  label: null,
  error: false,
  message: '',
  success: false,
  warning: false,
  values: null,
  initialValues: null,
  onChange: () => {},
};

CheckboxField.Group = CheckboxFieldGroup;

export default CheckboxField;
