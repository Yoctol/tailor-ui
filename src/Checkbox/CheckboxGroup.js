import PropTypes from 'prop-types';
import React from 'react';
import { Set } from 'react-powerplug';

import Flex from '../Grid/Flex';

import Checkbox from './Checkbox';

const CheckboxGroup = ({
  values: groupValues,
  initialValues,
  options,
  onChange,
  direction,
  ...otherProps
}) => (
  <Flex flexDirection={direction === 'horizontal' ? 'row' : 'column'}>
    <Set initial={initialValues || []} onChange={onChange}>
      {({ add, remove, has }) =>
        options.map(({ label, value, disabled = false }, index) => {
          const spacing = {
            [direction === 'horizontal' ? 'mr' : 'mb']:
              index + 1 !== options.length ? 2 : 0,
          };

          return (
            <Checkbox
              key={label}
              checked={groupValues ? groupValues.includes(value) : has(value)}
              disabled={disabled}
              onChange={event => {
                const { checked } = event.target;
                if (groupValues) {
                  if (groupValues.includes(value) && !checked) {
                    onChange(groupValues.filter(val => val !== value));
                  }
                  if (!groupValues.includes(value) && checked) {
                    onChange([...groupValues, value]);
                  }
                } else {
                  if (has(value) && !checked) {
                    remove(value);
                  }
                  if (!has(value) && checked) {
                    add(value);
                  }
                }
              }}
              {...otherProps}
              {...spacing}
            >
              {label}
            </Checkbox>
          );
        })
      }
    </Set>
  </Flex>
);

CheckboxGroup.displayName = 'CheckboxField.Group';

CheckboxGroup.propTypes = {
  /**
   * to specify the direction of the step bar, `horizontal` and `vertical` are currently supported
   */
  direction: PropTypes.oneOf(['horizontal', 'vertical']),
  /**
   * To set the initial values
   */
  initialValues: PropTypes.arrayOf(PropTypes.string),
  /**
   * Specifies options
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
      disabled: PropTypes.bool,
    })
  ).isRequired,
  /**
   * Used for setting the currently selected value
   */
  values: PropTypes.arrayOf(PropTypes.string),
  /**
   * The callback function that is triggered when the state changes
   */
  onChange: PropTypes.func,
};

CheckboxGroup.defaultProps = {
  direction: 'horizontal',
  values: null,
  initialValues: null,
  onChange: () => {},
};

export default CheckboxGroup;
