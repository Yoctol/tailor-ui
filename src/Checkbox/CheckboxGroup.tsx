import React, { SFC } from 'react';
import { Set } from 'react-powerplug';

import styled from 'utils/styled-components';

import Checkbox, { Direction } from './Checkbox';
import { Provider } from './CheckboxContext';

const CheckboxGroupFlex = styled<{ direction: Direction }, 'div'>('div')`
  display: ${p => (p.direction === 'horizontal' ? 'flex' : 'inline-flex')};
  flex-direction: ${p => (p.direction === 'horizontal' ? 'row' : 'column')};
`;

export interface CheckboxGroupProps {
  /**
   * to specify the direction of the step bar, `horizontal` and `vertical` are currently supported
   */
  direction?: Direction;
  /**
   * To set the initial value
   */
  defaultValue?: string[];
  /**
   * Specifies options
   */
  options?: {
    label: string;
    value: string;
    disabled?: boolean;
  }[];

  /**
   * Used for setting the currently selected value
   */
  value?: string[];
  /**
   * The callback function that is triggered when the state changes
   */
  onChange?: (value: ReadonlyArray<string>) => void;
}

const CheckboxGroup: SFC<CheckboxGroupProps> = ({
  value,
  defaultValue,
  options = null,
  onChange,
  direction = 'horizontal',
  children,
  ...otherProps
}) => (
  <CheckboxGroupFlex direction={direction}>
    <Set
      initial={defaultValue || []}
      onChange={_value => {
        if (onChange && _value) {
          onChange(_value);
        }
      }}
    >
      {({ add, remove, has }) => (
        <Provider
          value={{
            direction,
            _onChange: (event, _value) => {
              const { checked } = event.target;

              if (value) {
                if (value.includes(_value) && !checked && onChange) {
                  onChange(value.filter(val => val !== _value));
                }
                if (!value.includes(_value) && checked && onChange) {
                  onChange([...value, _value]);
                }
              } else {
                if (has(_value) && !checked) {
                  remove(_value);
                }
                if (!has(_value) && checked) {
                  add(_value);
                }
              }
            },
            _isChecked: _value =>
              value ? value.includes(_value) : has(_value),
          }}
        >
          {options
            ? options.map(({ label, value: optionValue, disabled = false }) => (
                <Checkbox
                  key={label}
                  value={optionValue}
                  disabled={disabled}
                  {...otherProps}
                >
                  {label}
                </Checkbox>
              ))
            : children}
        </Provider>
      )}
    </Set>
  </CheckboxGroupFlex>
);

CheckboxGroup.displayName = 'Checkbox.Group';

export default CheckboxGroup;
