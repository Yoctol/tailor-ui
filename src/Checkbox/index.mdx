---
name: Checkbox
menu: Data Entry
route: /data-entry/checkbox
---

import { PropsTable, Playground } from 'docz';
import { Toggle, Value } from 'react-powerplug';
import FormField from '../Form/FormField';
import Label from '../Form/Label';
import Button from '../Button';
import Checkbox from '.';
import CheckboxField, { CheckboxFieldGroup } from './CheckboxField';


# Checkbox

Checkbox component.

## When To Use

- Used for selecting multiple values from several options.
- If you use only one checkbox, it is the same as using Switch to toggle between two states. The difference is that Switch will trigger the state change directly, but Checkbox just marks the state as changed and this needs to be submitted.

## Examples

```js
import { FormField, Label, Checkbox, CheckboxField } from '@yoctol/ui';
```

### Checkbox

#### Basic

<Playground>
  <Toggle>
    {({ on, toggle }) => (
      <FormField>
        <Checkbox
          id="checkbox-basic"
          checked={on}
          onChange={toggle}
        />
        <Label htmlFor="checkbox-basic">Checkbox</Label>
      </FormField>
    )}
  </Toggle>
</Playground>

#### With disabled

<Playground>
  <FormField>
    <Checkbox id="checked-disabled" checked disabled />
    <Label htmlFor="checked-disabled">Checked Disabled</Label>
  </FormField>
  <FormField>
    <Checkbox id="disabled" disabled />
    <Label htmlFor="disabled">Disabled</Label>
  </FormField>
</Playground>

#### Customize color

<Playground>
  <Toggle>
    {({ on, toggle }) => (
      <FormField>
        <Checkbox
          id="checkbox-custom"
          bg="#63bf2d"
          borderColor="#423b63"
          checked={on}
          onChange={toggle}
        />
        <Label htmlFor="checkbox-custom">Checkbox</Label>
      </FormField>
    )}
  </Toggle>
</Playground>

### CheckboxField

#### Basic

<Playground>
  <Toggle>
    {({ on, toggle }) => (
      <CheckboxField checked={on} onChange={toggle}>
        Checkbox
      </CheckboxField>
    )}
  </Toggle>
</Playground>

#### With error

<Playground>
  <Toggle>
    {({ on, toggle }) => (
      <CheckboxField
        checked={on}
        onChange={toggle}
        error
        message="Something went wrong!"
      >
        Checkbox
      </CheckboxField>
    )}
  </Toggle>
</Playground>

### CheckboxField.Group

#### Basic

<Playground>
  <Value initial={['check_1']}>
    {({ set, value }) => (
      <CheckboxField.Group
        label="Checkbox Group"
        options={[
          { label: 'checkbox 1', value: 'check_1' },
          { label: 'checkbox 2', value: 'check_2' },
          { label: 'checkbox 3', value: 'check_3', disabled: true },
          { label: 'checkbox 4', value: 'check_4' },
        ]}
        values={value}
        onChange={set}
      />
    )}
  </Value>
</Playground>

#### With group & error

<Playground>
  <Value initial={[]}>
    {({ set, value }) => (
      <CheckboxField.Group
        label="Checkbox Group"
        error
        message="Something went wrong!"
        options={[
          { label: 'checkbox 1', value: 'check_1' },
          { label: 'checkbox 2', value: 'check_2' },
          { label: 'checkbox 3', value: 'check_3' },
          { label: 'checkbox 4', value: 'check_4' },
        ]}
        values={value}
        onChange={val => set(val)}
      />
    )}
  </Value>
</Playground>

#### With controlled

<Playground>
  <Value initial={[]}>
    {({ set, value }) => (
      <div>
        <CheckboxField.Group
          label="Checkbox Group"
          options={[
            { label: 'checkbox 1', value: 'check_1' },
            { label: 'checkbox 2', value: 'check_2' },
            { label: 'checkbox 3', value: 'check_3' },
            { label: 'checkbox 4', value: 'check_4' },
          ]}
          values={value}
          onChange={set}
        />
        <Button onClick={() => set(['check_1', 'check_2', 'check_3', 'check_4'])}>Select All</Button>
        <Button ml="2" onClick={() => set([])}>Unselect All</Button>
      </div>
    )}
  </Value>
</Playground>


## API

### Checkbox

<PropsTable of={Checkbox} />

### CheckboxField

<PropsTable of={CheckboxField} />

### CheckboxField.Group

<PropsTable of={CheckboxFieldGroup} />
