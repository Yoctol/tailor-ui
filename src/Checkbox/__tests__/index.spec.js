import React from 'react';

import { render } from 'test/test-utils';

import Checkbox from '..';

describe('Checkbox', () => {
  it('should render correctly', () => {
    const { container } = render(<Checkbox>Checkbox</Checkbox>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with props disabled', () => {
    const { container } = render(<Checkbox disabled>Checkbox</Checkbox>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
