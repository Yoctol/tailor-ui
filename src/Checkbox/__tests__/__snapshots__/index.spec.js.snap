// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `
.c7 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c3 {
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
}

.c6 {
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: 0.0625rem solid;
  border-radius: 2px;
  border-color: #c3c3c3;
  background-color: #ffffff;
  color: #444444;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c6::after {
  content: '';
  position: absolute;
  top: 1px;
  left: 5px;
  width: 5px;
  height: 10px;
  border: solid white;
  border-width: 0 2px 2px 0;
  opacity: 0;
  -webkit-transform: rotate(45deg) scale(0);
  -ms-transform: rotate(45deg) scale(0);
  transform: rotate(45deg) scale(0);
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c4 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
}

.c4:checked + .c5 {
  border-color: #423b63;
  background-color: #423b63;
}

.c4:checked + .c5::after {
  opacity: 1;
  -webkit-transform: rotate(45deg) scale(1);
  -ms-transform: rotate(45deg) scale(1);
  transform: rotate(45deg) scale(1);
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c2 + .c2 {
  margin-top: 0.5rem;
}

.c2:hover .c5 {
  border-color: #423b63;
}

.c0 + .c1 {
  margin-top: 0.5rem;
}

<label
  class="c0 c1 c2"
>
  <span
    class="c3"
  >
    <input
      class="c4"
      type="checkbox"
    />
    <span
      class="c5 c6"
    />
  </span>
  <div
    class="c7"
  >
    Checkbox
  </div>
</label>
`;

exports[`Checkbox should render with props disabled 1`] = `
.c7 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c3 {
  display: inline-block;
  position: relative;
  margin-bottom: 1px;
  line-height: 1;
}

.c6 {
  display: block;
  position: relative;
  width: 16px;
  height: 16px;
  padding: 0;
  border: 0.0625rem solid;
  border-radius: 2px;
  border-color: #c3c3c3;
  background-color: #ffffff;
  color: #444444;
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c6::after {
  content: '';
  position: absolute;
  top: 1px;
  left: 5px;
  width: 5px;
  height: 10px;
  border: solid white;
  border-width: 0 2px 2px 0;
  opacity: 0;
  -webkit-transform: rotate(45deg) scale(0);
  -ms-transform: rotate(45deg) scale(0);
  transform: rotate(45deg) scale(0);
  -webkit-transition: all 200ms ease 0s;
  transition: all 200ms ease 0s;
}

.c4 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
}

.c4:checked + .c5 {
  border-color: #423b63;
  background-color: #423b63;
}

.c4:checked + .c5::after {
  opacity: 1;
  -webkit-transform: rotate(45deg) scale(1);
  -ms-transform: rotate(45deg) scale(1);
  transform: rotate(45deg) scale(1);
}

.c8:hover .c5 {
  border-color: #423b63;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
  opacity: 0.5;
}

.c2 + .c2 {
  margin-top: 0.5rem;
}

.c0 + .c1 {
  margin-top: 0.5rem;
}

<label
  class="c0 c1 c2"
  disabled=""
>
  <span
    class="c3"
  >
    <input
      class="c4"
      disabled=""
      type="checkbox"
    />
    <span
      class="c5 c6"
    />
  </span>
  <div
    class="c7"
  >
    Checkbox
  </div>
</label>
`;
